test/functional/readpreference.test.js

```diff
@@ -78,6 +78,50 @@ describe('ReadPreference', function () {
     });
   });
 
+  describe('fromOptions factory method', () => {
+    const { PRIMARY, SECONDARY } = ReadPreference;
+    const TAGS = [{ loc: 'dc' }];
+
+    it('should return undefined if no options are passed', () => {
+      const readPreference = ReadPreference.fromOptions();
+      expect(readPreference).to.be.undefined;
+    });
+
+    context('readPreference is string', () => {
+      it('should accept { readPreference }', function () {
+        const readPreference = ReadPreference.fromOptions({
+          readPreference: PRIMARY
+        });
+        expect(readPreference).to.be.an.instanceOf(ReadPreference);
+        expect(readPreference.mode).to.equal(PRIMARY);
+      });
+
+      it('should accept { readPreference, readPreferenceTags }', function () {
+        const readPreference = ReadPreference.fromOptions({
+          readPreference: SECONDARY,
+          readPreferenceTags: TAGS
+        });
+        expect(readPreference).to.be.an.instanceOf(ReadPreference);
+        expect(readPreference.mode).to.equal(SECONDARY);
+        expect(readPreference.tags).to.eql(TAGS);
```

Comment:
Let's make the assertions consistent, with a preference for `equal`
In fact, its more of a stylistic opinion but I usually go with property checks:
```js
expect(readPreference).to.have.property('mode', SECONDARY);
```
this way if it fails it prints out a diff of the whole object. But sticking with `equal` is ok too!

edit:
Ok TIL that `eql` is deep equal :) facinating I made it this far without knowing, we've just used `.to.be.deep.equal(xx)`. I don't have a strong opinion on using one or the other.